test/tests/functional/pbs_job_array.py

```diff
@@ -715,14 +717,20 @@ def test_subjob_wrong_state(self):
         # while the server is sending the jobs to the MoM, restart the server
         self.server.restart()
         # make sure the mom is free so the scheduler can run jobs on it
-        self.server.expect(NODE, {'state': 'free'}, id=self.mom.shortname)
-        self.logger.info('Sleeping to ensure licenses are received')
-        time.sleep(5)
-        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
-                            {'scheduling': 'True'})
-        # ensure the sched cycle is finished
-        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
-                            {'scheduling': 'False'})
+        # if mom is in job-busy state all jobs are already scheduled.
+        attr = {'state': (MATCH_RE, 'free|job-busy')}
+        self.server.expect(NODE, attr, self.mom.shortname)
+        n_state = self.server.status(NODE, 'state',
+                                     id=self.mom.shortname)[0]['state']
+        # triggering scheduling cycle all jobs are in R state.
+        if n_state == 'free':
+            self.logger.info('Sleeping to ensure licenses are received')
+            time.sleep(5)
```

Comment:
What licenses are we talking about here?  If we're talking about PBS licenses, it's going to hang the server while they are being gathered.  The status() will hang.  I don't think we need to sleep.